controllers/k8ssandra/cleanup_test.go

```diff
@@ -0,0 +1,102 @@
+package k8ssandra
+
+import (
+	"context"
+	"testing"
+
+	testlogr "github.com/go-logr/logr/testing"
+	k8ssandraapi "github.com/k8ssandra/k8ssandra-operator/apis/k8ssandra/v1alpha1"
+	k8ssandralabels "github.com/k8ssandra/k8ssandra-operator/pkg/labels"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestK8ssandraClusterReconciler_DeleteServices(t *testing.T) {
```

Comment:
These tests are nicely structured. Good to see them in here as unit tests rather than standing up the whole envtest framework.